### DISCUSSION TITLE
How can we get the indices of examples having class overlap, when using the find_overlapping_classes method?

### DISCUSSION THREAD

**Discussion by rishabh706** [cleanlab/cleanlab#839] [Q&A]
- How can we get the indices of examples having class overlap? when using the find_overlapping_classes method, so that we can merge only those examples.
<img width="794" alt="Screenshot 2023-09-04 at 12 25 29 PM" src="https://github.com/cleanlab/cleanlab/assets/37527532/43063013-89b5-4f33-b120-aefad45b29a1">

**Comment by elisno** [comment]
Hi @rishabh706, thank you for the question!

It seems you're referring to the [_Find Dataset-level Issues for Dataset Curation_ tutorial](https://docs.cleanlab.ai/v2.4.0/tutorials/dataset_health.html#Start-of-tutorial:-Evaluate-the-health-of-8-popular-datasets), so I'll use that as a reference.

Instead of running the loop over all the datasets, I just ran the the following for the Caltech256 dataset. 

```python
# load class names, given labels, and predicted probabilities from already-trained model
pred_probs, labels, class_names = _load_classes_predprobs_labels("caltech256")

df = cleanlab.dataset.find_overlapping_classes(labels, pred_probs, class_names=class_names)
```


`find_overlapping_classes` returns an estimated count of overlapping examples between pairs of classes. However, it doesn't specifically track which examples are considered overlapping. Therefore, to fetch all examples that belong to either of the overlapping class indices, we can use the following method:

Given that you have the `labels` array and the `df` dataframe, here's how you can get the indices of examples belonging to overlapping classes:

```python
# Extract the row or rows of interest from the df
selected_rows = df.iloc[:1]  # This gets the first row. Adjust the slice as needed for multiple rows.

# Extract class indices
class_index_A = selected_rows["Class Index A"].values
class_index_B = selected_rows["Class Index B"].values

# Find overlapping indices in labels for both classes
overlap_indices_A = np.where(labels[:, None] == class_index_A)[0]
overlap_indices_B = np.where(labels[:, None] == class_index_B)[0]

# Combine the indices to get all examples of the overlapping classes
overlap_indices = np.unique(np.concatenate((overlap_indices_A, overlap_indices_B)))

print(overlap_indices)
```

This will provide you with the indices of ALL examples that belong to the overlapping classes, not just the ones estimated to be overlapping.

**Comment by rishabh706** [comment]
Hi @elisno, 
Thank you very much for the quick response. However is it possible to get the indices of only the overlapping examples, for each pair of overlapping classes? For example:- in the below image, the num_overlapping_examples for (corn, ear) pair are 44. Is there any way we can get the indices of only those 44 examples?
<img width="798" alt="Screenshot 2023-09-05 at 10 58 30 AM" src="https://github.com/cleanlab/cleanlab/assets/37527532/2cc8fff0-9bfe-41cb-9329-801804cae61f">
